### PR TITLE
quich: init at 3.1.0

### PR DESCRIPTION
**Pull request by xdHampus**
###### Motivation for this change
Wanted a calculator that was somewhat capable but easy to use.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/quich/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "quich";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/quich/default.nix:4]:
```suggestion
  pname = "quich";
```

File: pkgs/tools/misc/quich/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "quich"; 
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Usbac";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/quich/default.nix:9]:
```suggestion
    repo = pname;
```

File: pkgs/tools/misc/quich/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    rev = "v${version}";
+    sha256 = "1n9c01q2v6g9wnmxp248yclhp8cxclnj0yyn1qrvjsn6srcpr22c";
+  };
+
+  doCheck = true;
+
+  installPhase = "
+    mkdir -p $out/bin
+    make
+    cp quich $out/bin/quich
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/quich/default.nix:19]:
```suggestion
    cp quich $out/bin/quich
```

File: pkgs/tools/misc/quich/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "1n9c01q2v6g9wnmxp248yclhp8cxclnj0yyn1qrvjsn6srcpr22c";
+  };
+
+  doCheck = true;
+
+  installPhase = "
+    mkdir -p $out/bin
+    make
+    cp quich $out/bin/quich      
+  ";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/quich/default.nix:20]: Please use makeFlags.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -30702,4 +30702,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  quich = callPackage ../tools/misc/quich { } ;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:30706]: Please sort this entry.

File: pkgs/tools/misc/quich/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    description = "The advanced terminal calculator";
+    longDescription = ''
+      Quich is a compact, fast, powerful and useful calculator for your terminal
+      with numerous features, supporting Windows and Linux Systems,
+      written in ANSI C.
+    '';
+    homepage = "https://github.com/Usbac/quich";
+    license = licenses.mit;
+    maintainers = [ maintainers.xdhampus ];
+    platforms = platforms.linux ++ platforms.windows;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/quich/default.nix:32]:
```suggestion
    platforms = platforms.all.
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -11052,4 +11052,10 @@
     github = "zupo";
     githubId = 311580;
   };
+  xdhampus = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11055]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118286` at 217fedd7 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>quich</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/quich/default.nix:16:3:
```
   |
16 |   installPhase = "
   |   ^
```
</li>
  </ul>
</details>

**Comment by risicle** [comment]
Fails to build, macos 10.14:

```
  unpacking source archive /nix/store/kh5jxp3gxlfxqjgl4b2m4xnha9i869ih-source
  source root is source
  patching sources
  configuring
  no configure script, doing nothing
  building
  build flags: SHELL=/nix/store/74shlfgb18717ixjlpivpxd7iqcyhyn5-bash-4.4-p23/bin/bash DESTDIR= PREFIX=\$\(out\)
  gcc -o quich src/helper.c src/parser.c src/lexer.c src/variable.c src/quich.c -lm -std=c99 -pedantic -Wall -Wextra -Wmissing-prototypes -Wstrict-prototypes -Wold-style-definition 
  /nix/store/74shlfgb18717ixjlpivpxd7iqcyhyn5-bash-4.4-p23/bin/bash: gcc: command not found
  make: *** [Makefile:13: quich] Error 127
```

We either need to patch `${stdenv.cc.targetPrefix}cc` in there or disable for darwin.

File: pkgs/tools/misc/quich/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "Usbac";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1n9c01q2v6g9wnmxp248yclhp8cxclnj0yyn1qrvjsn6srcpr22c";
+  };
+
+  doCheck = true;
+
+  makeFlags = [ "DESTDIR=" "PREFIX=$(out)" ];
```
Comment (risicle) [inline pkgs/tools/misc/quich/default.nix:16]:
```suggestion
  makeFlags = [ "DESTDIR=" "PREFIX=$(out)"  "CC=${stdenv.cc.targetPrefix}cc"  ];
```

Fixes build for me, macos 10.14
